src/Model/Actions.re

```diff
@@ -126,6 +126,10 @@ type t =
   | Sneak(Sneak.action)
   | PaneTabClicked(Pane.paneType)
   | VimDirectoryChanged(string)
+  | WindowCloseBlocked
+  | WindowCloseDiscardConfirmed
+  | WindowCloseSaveAllConfirmed
```

Comment:
These are a bit awkward as they're essentially user intents. I also don't think it's necessarily a problem to have "action-like" messages, and with feature projects the encapsulation and lack of granularity might sometimes make event-like messages difficult. But enforcing event-like names is still very useful to stop thinking of them as functions.

This is also a reason why I'm leaning more towards using `msg` instead of `event`. Another reason being that the term is established and familiar from the Elm architecture.

Reply:
Sounds good! It seems like we're converging towards a very Elm-like architecture (ie, `updater`, `subscription`, and `msg`).